### PR TITLE
Simulate submit clearer responces

### LINKED ISSUE #44: Bouquet should give better indication of bundle landing

### PR DESCRIPTION
**Pull request by 0xjimmy**
- Fixed issue with blocks resetting to 0

Resolves #44 
- Made simulation results clearer
- Fix display of pending and mined bundle submissions. Show tx hashes of txes in bundle with link to explorers if on known network (Mainnet/Goerli)

### CODE REVIEW

File: app/ts/components/Submit.tsx [hunk 1]
```diff
@@ -28,74 +36,73 @@ const SimulationResult = ({
...
 							<span class='w-16 text-right'>Gas Used</span>
-							<span class='rounded bg-background px-2 py-1 font-mono font-medium'>{state.value.value.firstRevert.gasUsed} gas</span>
+							<span class='bg-black px-2 py-1 font-mono font-medium'>{state.value.value.firstRevert.gasUsed} ETH</span>
 						</div>
 						<div class='flex gap-2 items-center'>
 							<span class='w-16 text-right'>Error</span>
-							<span class='rounded bg-background px-2 py-1 font-mono font-medium'>
-								{'error' in state.value.value.firstRevert ? String(state.value.value.firstRevert.error) : 'Unknown'}
-							</span>
+							<span class='bg-black px-2 py-1 font-mono font-medium'>{'error' in state.value.value.firstRevert ? String(state.value.value.firstRevert.error) : 'Unknown'}</span>
```
Comment (KillariDev) [inline app/ts/components/Submit.tsx:61]: can we get rid of casting to String?

File: app/ts/library/provider.ts [hunk 2]
```diff
@@ -69,6 +69,11 @@ export const connectBrowserProvider = async (
 
 	const provider = new BrowserProvider(window.ethereum, 'any')
 
+	const blockCallback = async (blockNumber: number) => {
+		const block = await provider.getBlock(blockNumber)
+		if (block) updateLatestBlock(block, store, blockInfo, signers)
```
Comment (KillariDev) [inline app/ts/library/provider.ts:74]:
```suggestion
		if (block) await updateLatestBlock(block, store, blockInfo, signers)
```

File: app/ts/library/provider.ts [hunk 3]
```diff
@@ -100,15 +105,11 @@ export const connectBrowserProvider = async (
...
-	}
-	const blockCallback = (block: Block | null) => {
 		if (block) updateLatestBlock(block, store, blockInfo, signers)
 	}
 
-	provider.getBlock('latest').then((block) => {
-		if (block) updateLatestBlock(block, store, blockInfo, signers)
-	})
+	const block = await provider.getBlock('latest')
+	if (block) updateLatestBlock(block, store, blockInfo, signers)
```
Comment (KillariDev) [inline app/ts/library/provider.ts:112]:
```suggestion
	if (block) await updateLatestBlock(block, store, blockInfo, signers)
```

**Review by KillariDev — APPROVED** [review]

File: app/ts/constants.ts [hunk 4]
```diff
@@ -1,2 +1,7 @@
 export const MEV_RELAY_MAINNET = 'https://relay.dark.florist/'
 export const MEV_RELAY_GOERLI = 'https://relay-goerli.dark.florist/'
+
+export const BLOCK_EXPLORERS: { [chainId: string]: string } = {
+	'1': 'https://etherscan.io/',
+	'5': 'https://goerli.etherscan.io/'
+}
```
[the remark this answers is not in the document]
Comment (0xjimmy) [reply app/ts/constants.ts:7]: I don't believe we have a flashbots proxy for Sepolia.

File: app/ts/library/provider.ts (hunk 2 above)
Comment (0xjimmy) [reply app/ts/library/provider.ts:74]: No need to await, as there is nothing else to execute after in the callback

File: app/ts/components/Submit.tsx [hunk 5]
```diff
@@ -17,7 +19,13 @@ type PendingBundle = {
 			included: boolean
 		}
 	}
-	error?: Error
+	error?: Error,
+	success?: {
```
[the remark this answers is not in the document]
Comment (0xjimmy) [reply app/ts/components/Submit.tsx:23]: Kinda annoying to refactor that out, so I'll add it as low priority issue

**Pull request merged**